src/components/fields/date-input/date-input.tsx

```diff
@@ -0,0 +1,122 @@
+import { DateTimeFormatter, LocalDate, ResolverStyle } from "@js-joda/core";
+import { Locale } from "@js-joda/locale_en";
+import { Form } from "@lifesg/react-design-system/form";
+import { useEffect, useState } from "react";
+import { useFormContext } from "react-hook-form";
+import { useValidationSchema } from "src/utils/hooks";
+import * as Yup from "yup";
+import { IGenericFieldProps } from "../../frontend-engine/types";
+import { IDateInputSchema } from "./types";
+
+const INVALID_DATE = "Invalid date";
+export const DateInput = (props: IGenericFieldProps<IDateInputSchema>) => {
+	// =============================================================================
+	// CONST, STATE, REF
+	// =============================================================================
+	const {
+		schema: { label, useCurrentDate, dateFormat = "uuuu-MM-dd", validation, ...otherSchema },
+		id,
+		onChange,
+		value,
+		error,
+		...otherProps
+	} = props;
+	const [dateFormatter, setDateFormatter] = useState<DateTimeFormatter>();
+	const [stateValue, setStateValue] = useState<string>(value || ""); // adheres to dateFormat
+	const { setValue } = useFormContext();
+	const { setFieldValidationConfig } = useValidationSchema();
+
+	// =============================================================================
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
+		setFieldValidationConfig(
+			id,
+			Yup.string().test("is-date", "Invalid date", (value) => {
```

Comment:
note: `is-date` is a custom validation unique to date input field only, this is how we add custom validation on the component level and respect the approach to use Yup validation schema